n2survey/lime/survey.py

```diff
@@ -280,6 +291,203 @@ def get_responses(
+                    }
+                    subquestions = [inverted_answer_dict[answer] for answer in answers]
+                    condition_str = inner_logic.join(
+                        [f"{subquestion} == 'Y'" for subquestion in subquestions]
+                    )
+
+                # For other question types, e.g. "free", texts are left unchanged
+                else:
+                    condition_str = inner_logic.join(
+                        [f"{question} == {answer}" for answer in answers]
+                    )
+
+            # If only a single answer to question
+            else:
+                if question_type in ["single-choice", "array"]:
+                    inverted_answer_dict = {
+                        answer: id for id, answer in self.get_choices(question).items()
+                    }
+                    answers = inverted_answer_dict[answers]
+                    condition_str = f"{question} == '{answers}'"
+                elif question_type == "multiple-choice":
+                    inverted_answer_dict = {
+                        answer: id for id, answer in self.get_choices(question).items()
+                    }
+                    subquestion = inverted_answer_dict[answers]
+                    condition_str = f"{subquestion} == 'Y'"
+                else:
+                    condition_str = f"{question} == {answers}"
+            constructed_str = condition_str
+
+        return constructed_str
+
+    def keep_choices(self, conditions: Union[tuple, list]) -> "LimeSurvey":
+        """Filter responses DataFrame by keeping certain choices to specified questions
+
+        Args:
+            conditions (tuple or list of tuples): Conditions to re-format. Tuples in
+                format of (question_id, choices). choices can be a single str or list
+                of str. Multiple conditions are input as a list of tuples.
+                E.g. ("A6", "Woman") or [("A6", ["Woman", "Man"]), ("B6_SQ001", "Yes")]
+
+        Returns:
+            LimeSurvey: LimeSurvey with filtered responses
+        """
+
+        # Construct filter conditions str
+        constructed_filter_conditions = self.construct_filter_conditions(conditions)
+        # Make copy of LimeSurvey instance
+        filtered_survey = self._copy()
+        # Filter responses DataFrame
+        filtered_survey.responses = self.responses.query(constructed_filter_conditions)
+
+        return filtered_survey
+
+    def filter(self, mask: Union[pd.Series, pd.DataFrame]) -> "LimeSurvey":
+        """Filter responses DataFrame by applying a mask Series
+
+        Args:
+            mask (pd.Series): A bool Series or DataFrame as mask
+
+        Returns:
+            LimeSurvey: LimeSurvey with filtered responses
+        """
+
+        # Make copy of LimeSurvey instance
+        filtered_survey = self._copy()
+        # Filter responses DataFrame
+        filtered_survey.responses = self.responses[mask]
+
+        return filtered_survey
```

Comment:
What do you think if we use slicing instead of this function?

I mean some operations like: `servey[server.responses.A1 == "A1", "B1"]` will be allowed. 
It is done by defining `__getitem__` method. [Here](https://medium.com/@beef_and_rice/mastering-pythons-getitem-and-slicing-c94f85415e1c) you can find some details.

Basically, my idea is:
1) If we receive one argument (i.e. like `servey["B1"]`), we expect it to be the column names. So, we filter columns similar to "get_responses"
2) If we receive two arguments (i.e. like `servey[server.responses.A1 == "A1", "B1"]` or `servey[1:10, "B1"]`), then we do same as in 1 + filtering rows same as you did here.

The idea is still the same: make a copy + pass filters to the DataFrame object but the convenience is better I believe.